pkg/cmdline/cmdline.go

```diff
@@ -29,54 +25,27 @@ type CmdLine struct {
 	Err   error
 }
 
-var (
-	// procCmdLine package level static variable initialized once
-	once        sync.Once
-	procCmdLine CmdLine
-)
-
-func cmdLineOpener() {
-	cmdlineReader, err := os.Open("/proc/cmdline")
-	if err != nil {
-		errorMsg := fmt.Sprintf("Can't open /proc/cmdline: %v", err)
-		log.Print(errorMsg)
-		procCmdLine = CmdLine{Err: fmt.Errorf(errorMsg)}
-		return
-	}
-
-	procCmdLine = parse(cmdlineReader)
-	cmdlineReader.Close()
-}
-
 // NewCmdLine returns a populated CmdLine struct
-func NewCmdLine() CmdLine {
-	// We use cmdLineReader so tests can inject here
-	once.Do(cmdLineOpener)
-	return procCmdLine
+func NewCmdLine() *CmdLine {
```

Comment:
(Optional if you feel that this is okay to do given the other comments about breaking changes) If we are changing the function signature, maybe we should consider changing this function name too.

Consider a consumer of this library, they will have to call `cmdline.NewCmdLine()` which is redundant.  Given the context, it may be a positive change to rename this to `new` so consumers can call `cmdline.New()`?

Reply:
I think it's ok. I thought about this pretty hard. U-root programs that use this package took this change in stride, b/c Go nicely doesn't require you to do a declaration and will also nicely not require you to pick the -> dereference operator. This change should be transparent. 

I also much prefer making it clear it's a pointer on return: CmdLine has a map, and the map is NOT a value return. We had a recent bug where a struct returned by value had a pointer in it, where a pointer to struct return might have been clearer. 

Rust people are nodding their heads right now and saying "yep, we know."